flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/netty/TieredStorageResultSubpartitionView.java

```diff
@@ -94,14 +94,14 @@ public BufferAndBacklog getNextBuffer() throws IOException {
     @Override
     public AvailabilityWithBacklog getAvailabilityAndBacklog(int numCreditsAvailable) {
         if (findCurrentNettyPayloadQueue()) {
-            Queue<NettyPayload> currentQueue =
+            NettyPayloadQueue currentQueue =
                     nettyPayloadQueues.get(queueIndexContainsCurrentSegment);
+            checkState(numCreditsAvailable >= 0);
```

Comment:
This might already checked in `org.apache.flink.runtime.io.network.netty.CreditBasedSequenceNumberingViewReader#getNextBuffer`

Reply:
Removed.